cmd/peer-rest-server.go

```diff
@@ -1097,6 +1104,157 @@ func (s *peerRESTServer) GetPeerMetrics(w http.ResponseWriter, r *http.Request)
 	w.(http.Flusher).Flush()
 }
 
+type BenchmarkResult struct {
+	Uploads   uint64
+	Downloads uint64
+}
+
+func selfBenchmark(bucket string, size int, threads int, durationSecs int) (result BenchmarkResult, err error) {
+	fmt.Println("selfBenchmark")
+	objAPI := newObjectLayerFn()
+	if objAPI == nil {
+		return result, errServerNotInitialized
+	}
+
+	duration := time.Duration(durationSecs) * time.Second
+
+	buf := make([]byte, size)
+	rand.Read(buf)
```

Comment:
We shouldn't be allocating this kind of buffer instead have a reader that implements random bytes retuned

Reply:
```
func newRandomReader(seed, size int64) io.Reader {
        return io.LimitReader(rand.New(rand.NewSource(seed)), size)
}
```

You should just use this @krishnasrinivas